backend/src/main/java/com/google/rolecall/services/SectionServices.java

```diff
@@ -279,10 +279,40 @@ public Section editSection(SectionInfo newSection) throws EntityNotFoundExceptio
    *    in the database.
    */
   public void deleteSection(int id) throws EntityNotFoundException, InvalidParameterException {
+    checkDeleteOk(id, false);
+    Section section = getSection(id);
+
+    if(section.getType() == Section.Type.SUPER) {
+      // Check that all children are OK to delete
+      for(Position position: section.getPositions()) {
+        Integer childId = position.getSiblingId();
+        if(childId != 0 && childId > 0) {
```

Comment:
Seems like just `if (childId > 0)` is sufficient here?

Reply:
Fixed. Thanks!